src/services/qix/rpc.js

```diff
@@ -49,6 +50,39 @@ class RPC {
   }
 
   /**
+  * Reopens the connection and waits for the OnConnected notification.
+  * @param {Number} timeout - The time to wait for the OnConnected notification.
+  * @returns {Object} A promise containing the session state (SESSION_CREATED or SESSION_ATTACHED).
+  */
+  reopen(timeout) {
+    let timer;
+    let resolver;
+    const waitForNotification = new this.Promise((resolve) => { resolver = resolve; });
```

Comment:
I agree. Problem is that the time should not be set until we are connected (i.e. when the call to this.open() resolves). No excuse not make it more readable though, I'll try to work something out.